presto/eval.py

```diff
@@ -60,24 +62,81 @@ def __init__(
         years_to_remove: Optional[List[int]] = None,
         spatial_inference_savedir: Optional[Path] = None,
         seed: int = DEFAULT_SEED,
-        target_function: Optional[Callable[[Dict], int]] = None,
         filter_function: Optional[Callable[[pd.DataFrame], pd.DataFrame]] = None,
         name: Optional[str] = None,
         val_size: float = 0.2,
         dekadal: bool = False,
+        task_type: str = "cropland",
+        num_outputs: int = 1,
+        croptype_list: List = [],
+        finetune_classes: str = "CROPTYPE0",
+        downstream_classes: str = "CROPTYPE9",
+        balance: bool = False,
         train_masking: float = 0.0,
     ):
         self.seed = seed
-
-        if name is not None:
-            self.name = name
-        self.target_function = target_function
+        self.task_type = task_type
+        self.name = f"WorldCereal{task_type.title()}"
 
         train_data, val_data = WorldCerealLabelledDataset.split_df(train_data, val_size=val_size)
+
         self.train_df = prep_dataframe(train_data, filter_function, dekadal=dekadal)
         self.val_df = prep_dataframe(val_data, filter_function, dekadal=dekadal)
         self.test_df = prep_dataframe(test_data, filter_function, dekadal=dekadal)
 
+        if task_type == "cropland":
+            self.num_outputs = 1
+            self.croptype_list = []
+        elif task_type == "croptype":
+            # compress all classes in train that contain less
+            # than MIN_SAMPLES_PER_CLASS samples into "other"
```

Comment:
TBD, should we do it like this? We're creating a very diverse "other" class with many small exotic classes. Another option is to ignore these classes during training.